website2/src/app/docs/[slug]/page.tsx

```diff
@@ -1,21 +1,49 @@
 import { readFile, readdir } from "fs/promises";
-import { DocumentationHeader } from "@/components/documentationPage/documentationHeader";
+import { serialize } from "next-mdx-remote/serialize";
 import { Footer } from "@/components/footer";
 import { Header } from "@/components/header";
+import { Link, Sidebar } from "@/components/documentationPage/sidebar";
+import { DocumentationHeader } from "@/components/documentationPage/documentationHeader";
 import { Document } from "@/components/documentationPage/document";
+import styles from "./styles.module.css";
 
 const Page = async ({ params: { slug } }: { params: { slug: string } }) => {
   const source = await readFile(`documentation/${slug}.mdx`, "utf-8");
+  const links = await fetchLinks();
   return (
     <main>
       <Header />
-      <DocumentationHeader />
-      <Document source={source} />
+      <DocumentationHeader links={links} active={slug} />
+      <div className={styles.layout}>
+        <Sidebar links={links} active={slug} />
+        <Document source={source} />
+      </div>
       <Footer />
     </main>
   );
 };
 
+const fetchLinks = async (): Promise<Link[]> => {
+  const filenames = await readdir("documentation");
+  const metadata: Link[] = [];
+  for (let i = 0; i < filenames.length; i++) {
+    const file = await readFile(`documentation/${filenames[i]}`, "utf-8");
+    const meta = await serialize<
+      null,
+      { index: number; name: string; link: string }
+    >(file, {
+      parseFrontmatter: true,
+    });
+    const link = filenames[i].split(".mdx")[0];
+    metadata.push({
+      index: meta.frontmatter.index,
+      name: (meta.frontmatter.name as string) || link.split("_").join(" "),
+      link,
+    });
+  }
+  return metadata.sort((a, b) => a.index - b.index);
+};
```

Comment:
But what do you think about DRYing up the fetching of the docs and slug? This way we know the static pages and links are always in sync?

Up to you, TIOLI!

```suggestion
const getDocs = async (): Promise<Array<{ fileName: string, slug: string }>> => {
  const fileNames = await readdir("documentation");
  return fileNames.map((fileName) => ({
    fileName,
    slug: fileName.split(".mdx")[0],
  }));
}

const fetchLinks = async (): Promise<Link[]> => {
  const docs = await getDocs();
  const links = await Promise.all(docs.map(async ({ fileName, slug }) => {
    const file = await readFile(`documentation/${fileName}`, "utf-8");
    const meta = await serialize<
      null,
      { index: number; name: string; link: string }
    >(file, {
      parseFrontmatter: true,
    });
    return {
      index: meta.frontmatter.index,
      name: meta.frontmatter.name || slug.replaceAll("_", " "),
      link: slug,
    };
  }));
  return links.sort((a, b) => a.index - b.index);
};

export const generateStaticParams = async () => {
  return await getDocs();
};

```

Reply:
great idea, I'll make this change